src/theory/arith/arith_utilities.cpp

```diff
@@ -106,7 +106,11 @@ bool isTranscendentalKind(Kind k)
 
 Node getApproximateConstant(Node c, bool isLower, unsigned prec)
 {
-  Assert(c.isConst());
+  if (!c.isConst())
+  {
+    Assert(false) << "getApproximateConstant: non-constant input " << c;
+    return Node::null();
+  }
```

Comment:
I really don't like this. If we think that `c` should really be a constant here, we should assert this. If it is not, why would we make the production build robust to it?

Reply:
In theory, the code should always be robust to this (in both debug and production).  Moreover we shouldn't *require* a failure if we are in production, since this is an internal error.  This is the common use of "Assert" instead of "AlwaysAssert".